js/map/map-popup.js

```diff
@@ -0,0 +1,185 @@
+    element.setAttribute('src', dataObject.author.avatar);
+  } else { element.remove(); }
+};
+
+const createCardTitle = (element, dataObject) => {
+  if (dataObject.offer && dataObject.offer.title) {
+    element.textContent = dataObject.offer.title;
+  } else {
+    element.textContent = 'Заголовок предложения';
+    element.classList.add('visually-hidden');
+  }
+};
+
+const createCardAddress = (element, dataObject) => {
+  if (dataObject.offer && dataObject.offer.address) {
+    element.textContent = dataObject.offer.address;
+  } else { element.remove(); }
+};
+
+const createCardPrice = (element, dataObject) => {
+  if (dataObject.offer && dataObject.offer.price) {
+    element.textContent = dataObject.offer.price;
+    element.insertAdjacentHTML('beforeend', ` ${getPriceSuffix()}`);
+  } else { element.remove(); }
+};
+
+const createCardType = (element, dataObject) => {
+  if (dataObject.offer && dataObject.offer.type) {
+    element.textContent = getPlaceName(offerPlaceList, dataObject.offer.type);
+  } else {
+    element.textContent = 'Тип жилья не указан';
+    element.classList.add('visually-hidden');
+  }
+};
+
+const createCapacityDescription = (element, dataObject) => {
+  let guestsDescription = guestText.UNKNOWN_GUEST_TEXT;
+  if (dataObject.offer.guests) {
+    guestsDescription = getGuestDescription(dataObject.offer.guests);
+  }
+  let roomDescription = getRoomText().UNKNOWN_ROOM_TEXT;
+  if (dataObject.offer.rooms) {
+    roomDescription = getRoomDescription(dataObject.offer.rooms);
+  }
+  element.textContent = `${roomDescription} ${guestsDescription}`;
+};
+
+const createCardCapacity = (element, dataObject) => {
+  if (!dataObject.offer || (!dataObject.offer.rooms && !dataObject.offer.guests)) {
+    element.remove();
+  } else {
+    createCapacityDescription(element, dataObject);
+  }
+};
+
+const createCardCheckTime = (element, dataObject) => {
+  let checkinDescription = 'Время заезда не указано, ';
+  let checkoutDescription = 'время выезда не указано';
+
+  if (dataObject.offer && dataObject.offer.checkin) {
+    checkinDescription = `Заезд после ${dataObject.offer.checkin}, `;
+  }
+
+  if (dataObject.offer && dataObject.offer.checkout) {
+    checkoutDescription = `выезд до ${dataObject.offer.checkout}, `;
+  }
+  element.textContent = `${checkinDescription} ${checkoutDescription}`;
+};
+
+const createCardFeatures = (element, dataObject) => {
+  if (dataObject.offer && dataObject.offer.features) {
+    const featureList = dataObject.offer.features;
+    const featuresTemplateItems = element.querySelectorAll('.popup__feature');
+    const featureModifierList = featureList.map((feature) => `popup__feature--${feature}`);
+
+    featuresTemplateItems.forEach((featureItem) => {
+      const itemModifier = featureItem.classList[1];
+      if (!featureModifierList.includes(itemModifier)) {
+        featureItem.remove();
+      }
+    });
+  } else { element.remove(); }
+};
```

Comment:
функция - колбяк внутри `forEach` содержит в себе замыкание. Можешь выработать еще одну привычку.
создавать такие замыкания через фабрику - это в будущем позволит тестировать колбяк отдельно от цикла


```suggestion
const createCallbackClosureExample = (featureList)=>{
  const featureModifierList = featureList.map((feature) => `popup__feature--${feature}`);
  return (featureItem) => {
    const itemModifier = featureItem.classList[1];
    if (!featureModifierList.includes(itemModifier)) {
      featureItem.remove();
    }
  };
};

const createCardFeatures = (element, dataObject) => {
  if (dataObject.offer && dataObject.offer.features) {
    const featuresTemplateItems = element.querySelectorAll('.popup__feature');

    featuresTemplateItems.forEach(createCallbackClosureExample(dataObject.offer.features));
  } else {
    element.remove();
  }
};
```